src/server/transaction.h

```diff
@@ -84,6 +84,10 @@ using facade::OpStatus;
 //
 // ```
 
+// Check that the command can be run
+using KeyReadyChecker =
+    std::function<bool(EngineShard*, const DbContext& context, Transaction* tx, std::string_view)>;
+
```

Comment:
what if we move it into the transaction class next to WaitKeysProvider?

Reply:
Let's also extend the comment: `checks whether the touched key is valid for a blocking transaction watching it`